modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js

```diff
@@ -12,239 +12,145 @@
  * details.
  */
 
-import {wait} from '@testing-library/dom';
-import {act, cleanup, render} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import {cleanup, render} from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import {PageProvider} from 'data-engine-js-components-web';
 import moment from 'moment';
 import React from 'react';
 
 import DatePicker from '../../../src/main/resources/META-INF/resources/DatePicker/DatePicker.es';
 
-const spritemap = 'icons.svg';
-
-const defaultDatePickerConfig = {
-	locale: 'en_US',
-	localizedValue: {},
-	name: 'dateField',
-	spritemap,
-};
-
-const DatePickerWithProvider = (props) => (
-	<PageProvider value={{editingLanguageId: 'en_US'}}>
-		<DatePicker {...props} />
-	</PageProvider>
-);
-
 describe('DatePicker', () => {
-	// eslint-disable-next-line no-console
-	const originalWarn = console.warn;
-
-	beforeAll(() => {
-		// eslint-disable-next-line no-console
-		console.warn = (...args) => {
-			if (
-				/DataProvider: Trying/.test(args[0]) ||
-				/Deprecation warning: value provided is not in a recognized RFC2822 or ISO format/.test(
-					args[0]
-				)
-			) {
-				return;
-			}
-			originalWarn.call(console, ...args);
-		};
-	});
-
-	afterAll(() => {
-		// eslint-disable-next-line no-console
-		console.warn = originalWarn;
-	});
-
-	afterEach(cleanup);
-
 	beforeEach(() => {
-		jest.useFakeTimers();
 		fetch.mockResponseOnce(JSON.stringify({}));
 	});
 
+	afterEach(cleanup);
+
 	it('has a helptext', () => {
 		const {container} = render(
-			<DatePickerWithProvider
-				{...defaultDatePickerConfig}
-				tip="Type something"
-				value="06/02/2020"
-			/>
+			<DatePicker tip="Type something" value="06/02/2020" />
 		);
 
-		act(() => {
-			jest.runAllTimers();
-		});
-
 		expect(container.querySelector('.form-text')).toBeTruthy();
 	});
 
 	it('has a label', () => {
-		const {container} = render(
-			<DatePickerWithProvider
-				{...defaultDatePickerConfig}
-				label="label"
-			/>
-		);
-
-		act(() => {
-			jest.runAllTimers();
-		});
+		const {container} = render(<DatePicker label="label" />);
 
 		expect(container.querySelector('.ddm-label')).toBeTruthy();
 	});
 
 	it('has a predefinedValue', () => {
 		const {container} = render(
-			<DatePickerWithProvider
-				{...defaultDatePickerConfig}
-				predefinedValue="06/02/2020"
+			<DatePicker
+				label="date"
+				locale="en_US"
+				localizedValue={{}}
+				name="dateField"
+				predefinedValue="2020-06-02"
 			/>
 		);
 
-		act(() => {
-			jest.runAllTimers();
-		});
+		const inputDatePicker = container
+			.getElementsByClassName('form-control')
+			.item(0);
```

Comment:
Try to query element by role or label to ensure the component's accessibility, according to the [query priority](https://testing-library.com/docs/queries/about/#priority) 
```suggestion
		const inputDatePicker = getByRole('textbox');
```

note that if many items are being returned you should try to make the query more specific and as last resource use `queryAllBy...`

Reply:
Makes sense! Will be taking that in consideration next time.